### PR TITLE
JP-2802: Fix a bug in the definition of NIRSpec resampled WCS

### LINKED ISSUE #7265: Weird S_REGION for Spec2Pipeline resample_spec for 01539003001
_Issue [JP-2802](https://jira.stsci.edu/browse/JP-2802) was created on JIRA by [Brian Hayden](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bhayden):_

A help desk ticket came in, and [David Rodriguez](https://jira.stsci.edu/secure/ViewProfile.jspa?name=drodriguez) noticed some strange footprints for, specifically, jw01539003001_03102_00005_nrs2 and jw01539003001_03102_00001_nrs1. They seemed to span from one celestial pole to the other (-89 to 89).

I tracked down that jw01539003001_03102_00001_nrs1 is a member of jw01539-o003_20220720t061905_spec2_005, and checked the SDP log (attached), noticing the offending S_REGION occurs in the stpipe.Spec2Pipeline.resample_spec section.

 
```
2022-07-20 06:41:10,802 - stpipe.Spec2Pipeline.resample_spec - INFO - Update S_REGION to POLYGON ICRS  81.180590749 -89.374247193 81.112666873 -89.374247193 81.112666873 89.303900795 81.180590749 89.303900795
```

### PR DESCRIPTION
**Pull request by mcara**
Resolves [JP-2802](https://jira.stsci.edu/browse/JP-2802)
Closes #7265

This PR fixes a bug in the definition of the NIRSpec resampled WCS. Fundamentally, the bug was that for the forward transformation the horizontal position in the slit was incorrect. This PR fixes horizontal position to 0 - center of the slit. I expect regression tests to fail.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

Regression test: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/469/

### CODE REVIEW

File: jwst/resample/resample_spec.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def build_nirspec_output_wcs(self, refmodel=None):
             wmean_l = d2s(*np.mean(bbox, axis=1))[2]
 
         # transform the weighted means into target RA/Dec
+        print(f"wmean_s, wmean_l: {wmean_s, wmean_l}")
```
Comment (hbushouse) [inline jwst/resample/resample_spec.py:134]: Need to remove this print statement

**Review by hbushouse — APPROVED** [review]
Looks good. Just need to remove a print statement that shouldn't be there.

File: jwst/resample/resample_spec.py (hunk 1 above)
Comment (mcara) [reply jwst/resample/resample_spec.py]: Done.

**Review by nden — APPROVED** [review]
LGTM

**Comment by hbushouse** [comment]
The CI failure appears to be unrelated, so I'm going to merge.

**Pull request merged**